Implement build-linux wrapper for ARM (as opposed to direct cmake)

**Describe the bug**
`./build-linux.sh build` will fail on ARM or AARCH64 because LINUX_ON_ARM is never set and `-sse2` is still part of the CXXFLAGS

**Please let us know your surge version**
1.7.1 (tag from github)

* Surge Version: 1.7.1
* Plugin type (VST2, 3 or AU): doesn't matter
* Bits (32/64): either

**To Reproduce**
Steps to reproduce the behavior:
1. Try to build on arm or aarch64 Linux: `./build-linux.sh build`

It fails:
```
Scanning dependencies of target surge-vst3-dll
[  2%] Building CXX object CMakeFiles/surge-vst3-dll.dir/src/common/dsp/effect/ConditionerEffect.cpp.o
[  2%] Building CXX object CMakeFiles/surge-vst3-dll.dir/src/common/dsp/effect/DistortionEffect.cpp.o
c++: error: unrecognized command line option ‘-msse2’
make[3]: *** [CMakeFiles/surge-vst3-dll.dir/build.make:126: CMakeFiles/surge-vst3-dll.dir/src/common/dsp/effect/ConditionerEffect.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
c++: error: unrecognized command line option ‘-msse2’
make[3]: *** [CMakeFiles/surge-vst3-dll.dir/build.make:139: CMakeFiles/surge-vst3-dll.dir/src/common/dsp/effect/DistortionEffect.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:274: CMakeFiles/surge-vst3-dll.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:308: CMakeFiles/Surge-VST3-Packaged.dir/rule] Error 2
make: *** [Makefile:229: Surge-VST3-Packaged] Error 2
```

**Expected behavior**
It builds

**Screenshots**
n/a

**Desktop (please complete the following information):**
n/a

**Additional context**
The changelog explicitely say these arch are supported on Linux.
Maybe the linux script should just pass the proper option to cmake.

Thanks for all the insight.

Switching to only using cmake to build works on aarch64 (passing the right option). But my point still stand. Unless you are cross compiling, it should just detect the arch it is being build.

On arm, I get another error:
```
/usr/bin/c++  -DARM_NEON=1 -DLINUX=1 -DRELEASE=1 -DTARGET_VST3=1 -DUSE_STD_FILESYSTEM=1 -Dsurge_vst3_dll_EXPORTS -Ilibs -Ilibs/filesystem -Ilibs/xml -Ilibs/catch2 -Ilibs/strnatcmp -Ilibs/tuning-library/include -Isrc/common -Isrc/common/dsp -Isrc/common/thread -Isrc/common/vt_dsp -Igeninclude -Isrc/common/gui -Ivstgui.surge -Ilibs/nanosvg/src -Isrc/linux -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/arm-linux-gnueabihf/glib-2.0/include -I/usr/lib/arm-linux-gnueabihf/libffi-3.2.1/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -Ilintemp -Ilibs/simde -Isrc/vst3 -Ivst3sdk -Isurge-headless -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -O3 -DNDEBUG -fPIC   -march=native -mcpu=native -Wno-psabi -flax-vector-conversions "-D_aligned_malloc(x,a)=aligned_alloc(a,x)" "-D_aligned_free(x)=free(x)" -Wno-multichar -fvisibility=hidden -fvisibility-inlines-hidden -fdata-sections -ffunction-sections -std=c++17 -MD -MT CMakeFiles/surge-vst3-dll.dir/src/common/vt_dsp/lipol.cpp.o -MF CMakeFiles/surge-vst3-dll.dir/src/common/vt_dsp/lipol.cpp.o.d -o CMakeFiles/surge-vst3-dll.dir/src/common/vt_dsp/lipol.cpp.o -c src/common/vt_dsp/lipol.cpp
src/common/vt_dsp/lipol.cpp: In member function ‘void lipol_ps::set_blocksize(int)’:
src/common/vt_dsp/lipol.cpp:19:23: error: ‘vdupq_n_f32’ was not declared in this scope
   19 |    lipol_BLOCK_SIZE = vdupq_n_f32(bs);
      |                       ^~~~~~~~~~~
```

--- a/.gitignore
+++ b/.gitignore
@@ -79,6 +79,7 @@ cmake-stamp
 build/
 build32/
 buildlin/
+buildlin-*/
 buildmac/
 buildwin/
 build-arm/

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,7 @@ cmake_minimum_required(VERSION 3.10)
 set(CMAKE_OSX_DEPLOYMENT_TARGET 10.9 CACHE STRING "Build for 10.9")
 
 project(Surge VERSION 1.7.1.0 LANGUAGES CXX ASM)
+message( STATUS "CMake Version is ${CMAKE_VERSION}" )
 message( STATUS "Compiler Version is ${CMAKE_CXX_COMPILER_VERSION}" )
 
 add_custom_target(surge-shared)

--- a/README.md
+++ b/README.md
@@ -218,8 +218,9 @@ git submodule update --init --recursive
 
 `build-linux.sh` is a wrapper on the various cmake and make commands needed to build Surge. As with
 macOS, it is getting smaller every day as we move more things direclty into cmake. 
-As of this Surge 1.7.1, `build-linux.sh` only works on intel platforms. If you are building on 
-ARM please keep reading below for directions on the ARM compile mechanism.
+As of Surge 1.7.1, `build-linux.sh` only works on intel platforms. If you are building on 
+ARM, please use the ARM specific instructions below or use the head of the codebase, where
+we are continuing to improve the ARM experience and build-linux is ARM aware.
 
 You can build with the command:
 
@@ -261,6 +262,9 @@ make -j 2 Surge.vst3
 will build the VST3 and deposit it in surge/products.
 
 ## Building for ARM platforms
+
+As of August 4, build-linux supports ARM builds. If you are building the 1.7.0 or
+1.7.1 release, though, you need to follow these instructions.
 
 With 1.7.0 we have merged changes needed to build with ARM platforms and have done some 
 raspberry pi testing. Due to a variety of choices an ARM user needs to make, and due to

--- a/build-linux.sh
+++ b/build-linux.sh
@@ -37,6 +37,21 @@ GREEN=`tput setaf 2`
 NC=`tput sgr0`
 DEF_BUILD_DIR="buildlin"
 
+UNAME_M=`uname -m`
+if [[ "$UNAME_M" =~ ^arm ]]; then
+   ARM=1
+fi
+if [[ "$UNAME_M" =~ ^aarch ]]; then
+   ARM=1
+fi
+
+if [[ ! -z $ARM ]]; then
+   CMAKE_EXTRA_ARGS="-DARM_NATIVE=native"
+   DEF_BUILD_DIR="buildlin-${UNAME_M}"
+   echo "ARM build activated. Using ARM_NATIVE=native settings; building in ${DEF_BUILD_DIR}"
+fi
+
+
 prerequisite_check()
 {
     if [ ! -f vst3sdk/LICENSE.txt ]; then
@@ -51,7 +66,7 @@ prerequisite_check()
 
 run_cmake()
 {
-    cmake . -B${DEF_BUILD_DIR}
+    cmake . -B${DEF_BUILD_DIR} ${CMAKE_EXTRA_ARGS}
 }
 
 run_cmake_if()

--- a/src/common/vt_dsp/lipol.cpp
+++ b/src/common/vt_dsp/lipol.cpp
@@ -15,11 +15,7 @@ lipol_ps::lipol_ps()
 
 void lipol_ps::set_blocksize(int bs)
 {
-#ifdef ARM_NEON
-   lipol_BLOCK_SIZE = vdupq_n_f32(bs);
-#else
    lipol_BLOCK_SIZE = _mm_cvt_si2ss(m128_zero, bs);
-#endif
    m128_bs4_inv = _mm_div_ss(m128_four, lipol_BLOCK_SIZE);
 }
 

